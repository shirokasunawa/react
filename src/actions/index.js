export const addOne = () => ({
    type: 'ADD_ONE'
})
export const addX = x => ({
    type: 'ADD_X',
    payload: x
})

export const allStudents = students => ({
    type: 'GET_STUDENTS',
    payload: students
})