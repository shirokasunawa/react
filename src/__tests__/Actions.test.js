import configureMockStore from 'redux-mock-store'
import * as fromActions from '../actions'

// Create a mock store
const mockStore = configureMockStore()
const store = mockStore({})

describe('action creators', () => {
  it('creates ADD_X when add was successful', () => {
    store.dispatch(fromActions.addX(15));
    expect(store.getActions()).toMatchSnapshot();
  });
});