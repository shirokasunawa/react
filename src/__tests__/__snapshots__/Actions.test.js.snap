exports[`action creators creates ADD_X when add was successful 1`] = `
Array [
  Object {
    "payload": 15,
    "type": "ADD_X",
  },
]
`;
